libsrc/ParametricMapObject.cpp

```diff
@@ -0,0 +1,230 @@
+//
+// Created by Andrey Fedorov on 3/11/17.
+//
+
+#include <dcmqi/Helper.h>
+#include <itkMinimumMaximumImageCalculator.h>
+#include "dcmqi/ParametricMapObject.h"
+#include <dcmtk/dcmiod/iodcommn.h>
+#include <dcmtk/dcmpmap/dpmparametricmapiod.h>
+
+
+int ParametricMapObject::initializeFromITK(Float32ITKImageType::Pointer inputImage,
+                                           const string &metaDataStr,
+                                           std::vector<DcmDataset *> derivationDatasets) {
+  sourceRepresentationType = ITK_REPR;
+
+  itkImage = inputImage;
+
+  initializeMetaDataFromString(metaDataStr);
+
+  if(!metaDataIsComplete()){
+    updateMetaDataFromDICOM(derivationDatasets);
+  }
+
+  initializeVolumeGeometry();
+
+  // NB: the sequence of steps initializing different components of the object parallels that
+  //  in the original converter function. It probably makes sense to revisit the sequence
+  //  of these steps. It does not necessarily need to happen in this order.
+  initializeEquipmentInfo();
+  initializeContentIdentification();
+
+  // TODO: consider creating parametric map object after all FGs are initialized instead
+  createParametricMap();
```

Comment:
I think the method name should indicate the type (DICOM or ITK):

We could add two virtual methods {create/get}{DICOM/ITK}Representation to MultiframeObject that needs to be implemented by inheriting classes.